src/array.vala

```diff
@@ -41,11 +41,8 @@ public class Vast.Array : Object
     /* GObject that owns the memory buffer for storage of scalar elements */
     public Bytes? data {get; construct; }
 
-    /* global shift from the _cached_data; _cached_data + origin is the memory location of 0-th element */
-    public size_t origin { get; construct; default = 0; }
-
-    /* pointer to the begining memory buffer, */
-    internal unowned uint8* _cached_data;
+    /* pointer to the memory location of 0th element */
+    public uint8 * origin {get; construct; default = null;}
```

Comment:
For the property, we could have a public `size_t` and an internal computed pointer from get and construct without changing any name.